phac-nml/irida-next
# Groups/samples_count

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds a samples_count column to Groups to show the total number of samples contained within a group. This also includes all samples in projects within subgroups.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._
![image](https://github.com/user-attachments/assets/a51a7983-adbf-437e-8340-bfb55b39101a)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._
1. Sign in
2. Click on "Groups" on the sidebar
3. Find a group where you are the owner and take note of the number of samples listed (number next to the beaker icon)
4. Select the group
5. Look through "Subgroups and projects" and check that the number of samples is accumulating correctly as you move up from projects to subgroups or subgroups to the parent group
6. Compare the number of samples listed in step 3 and see that it equals the accumulated number from step 5
7. Select a project within that group
8. Click on "Samples" on the sidebar
9. Add a new sample to the project
10. Go back to the parent group and check to see that the number of samples increased by one
11. Repeat steps 8-10 but delete samples
12. Repeat steps 8-10 but clone samples
13. Repeat steps 8-10 but transfer samples
14. Click on "Projects" on the sidebar
15. Select a project where you are the owner
16. Select "Settings" -> "General" in the sidebar
17. Go to the Transfer project section and transfer the project to a different parent group
18. Click on "Projects" on the sidebar
19. Find the parent group and check that the sample counts are correct.
20. Go back into the same project and delete the project
21. Click on "Projects" on the sidebar
22. Find the parent group of the deleted project
23. Check that the sample counts are correct
24. Repeat steps 14-23 but transfer and delete a group instead

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
